File: macos/Classes/JsScriptPlugin.m
#import "JsScriptPlugin.h"

void jsContextSetup();

@implementation JsScriptPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"js_script"
            binaryMessenger:[registrar messenger]];
  JsScriptPlugin* instance = [[[JsScriptPlugin alloc] init] autorelease];
  [registrar addMethodCallDelegate:instance channel:channel];
    jsContextSetup();
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    result(FlutterMethodNotImplemented);
}

@end

File: macos/Classes/JsScriptPlugin.swift
import Cocoa
import FlutterMacOS

public class JsScriptPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "js_script", binaryMessenger: registrar.messenger)
    let instance = JsScriptPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
